framework/src/modules/bft/api.ts

```diff
@@ -156,6 +161,11 @@ export class BFTAPI extends BaseAPI {
 		certificateThreshold: bigint,
 		validators: Validator[],
 	): Promise<void> {
+		if (validators.length > this._batchSize) {
+			throw new Error(
+				`Invalid validators size. Validators must be greater than batch size ${this._batchSize}.`,
```

Comment:
I think the error message is incorrect (you want validators.length <= this._batchSize). It could be instead:

> `Invalid number of validators. The number of validators can be at most the batch size ${this._batchSize}.`,